src/models/chen2021Fitness/Chen2021FitnessWidget.tsx

```diff
@@ -0,0 +1,381 @@
+import { Widget } from '../../widgets/Widget';
+import { ZodQueryEncoder } from '../../helpers/query-encoder';
+import { SampleSelectorSchema } from '../../helpers/sample-selector';
+import * as zod from 'zod';
+import React, { useEffect, useState } from 'react';
+import { get } from '../../services/api';
+import {
+  Chen2021FitnessRequestSchema,
+  Chen2021FitnessResponseSchema,
+  ValueWithCISchema,
+} from './chen2021Fitness-types';
+import styled from 'styled-components';
+import { Col, Form, Button } from 'react-bootstrap';
+import { Plot } from '../../components/Plot';
+
+const ContainerPropsSchema = SampleSelectorSchema;
+type ContainerProps = zod.infer<typeof SampleSelectorSchema>;
+
+const ResultsPropsSchema = zod.object({
+  request: Chen2021FitnessRequestSchema,
+});
+type ResultsProps = zod.infer<typeof ResultsPropsSchema>;
+
+type Chen2021FitnessRequest = zod.infer<typeof Chen2021FitnessRequestSchema>;
+type Chen2021FitnessResponse = zod.infer<typeof Chen2021FitnessResponseSchema>;
+
+const getData = async (
+  params: Chen2021FitnessRequest,
+  signal: AbortSignal
+): Promise<Chen2021FitnessResponse | null> => {
+  const mutationsString = params.mutations.join(',');
+  const url =
+    `/computed/model/chen2021Fitness` +
+    `?country=${params.country}` +
+    `&mutations=${mutationsString}` +
+    `&matchPercentage=${params.matchPercentage}` +
+    (params.samplingStrategy ? `&dataType=${params.samplingStrategy}` : '') +
+    `&alpha=${params.alpha}` +
+    `&generationTime=${params.generationTime}` +
+    `&reproductionNumberWildtype=${params.reproductionNumberWildtype}` +
+    `&plotStartDate=${params.plotStartDate.toISOString().substring(0, 10)}` +
+    `&plotEndDate=${params.plotEndDate.toISOString().substring(0, 10)}` +
+    `&initialWildtypeCases=${params.initialWildtypeCases}` +
+    `&initialVariantCases=${params.initialVariantCases}`;
+  const response = await get(url, signal);
+  if (response.status !== 200) {
+    // The computation might fail, for example, if some values go out-of-bound. The issue shall be addressed with the
+    // introduction of a better error handling and reporting on the server side.
+    return null;
+  }
+  const data = await response.json();
+  if (!data) {
+    return null;
+  }
+  return Chen2021FitnessResponseSchema.parse(data);
+};
+
+const Wrapper = styled.div`
+  background-color: #ffe0b6;
+  padding: 15px;
+`;
+
+const SectionHeader = styled.h5`
+  margin-top: 20px;
+`;
+
+const Chen2021FitnessContainer = ({
```

Comment:
It's very hard to keep an overview of React code when multiple components are in one file. In some cases there are good reasons to put everything in one file, but in this case I think that the components can easily be put in separate files.